src/configuration-parser.ts

```diff
@@ -0,0 +1,32 @@
+import * as Ajv from 'ajv';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class ConfigurationParser {
+    private typewizConfig: any;
+
+    public parse(configurationPath: string) {
+        const typewizConfigSchema = JSON.parse(fs.readFileSync('src/typewiz.json', { encoding: 'utf8' }));
```

Comment:
While we do have sync IO in other places, it's generally where it's totally guaranteed that it's in the first tick (i.e. it's a CLI, or it's at the very start of a program running).

There's no such guarantee here, it might be better to use `util.promisify()` on `fs.readFile` and `async/await`.

Reply:
Right now ci runs against node 6.4.0 which doesn't have `util.promisify()` (async await should be transpiled by typescript). Is support for node 6 important or can we use `util.promisify()` (and then remove node 6 from ci since it wont work anymore.

Reply:
@urish Thoughts? Node 8.9.0 is the LTS, how far back do we want to support node versions?